lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1049,8 +1067,49 @@ instance Buildable ProtocolParameters where
         , "Desired number of pools: " <> build (pp ^. #desiredNumberOfStakePools)
         , "Minimum UTxO value: " <> build (pp ^. #minimumUTxOvalue)
         , "Eras:\n" <> indentF 2 (build (pp ^. #eras))
+        , "Execution unit prices: " <>
+            case (pp ^. #executionUnitPrices) of
+                Just prices -> build prices
+                Nothing -> "not specified"
         ]
 
+data ExecutionUnits = ExecutionUnits
+    { executionSteps
+        :: Word64
+        -- ^ This corresponds roughly to the time to execute a script.
+
+    , executionMemory
+        :: Word64
+        -- ^ This corresponds roughly to the peak memory used during script
+        -- execution.
+    } deriving (Eq, Generic, Show)
+
+data ExecutionUnitPrices = ExecutionUnitPrices
+    { priceExecutionSteps  :: Rational
+    , priceExecutionMemory :: Rational
+    } deriving (Eq, Generic, Show)
+
+instance NFData ExecutionUnitPrices
+
+instance Buildable ExecutionUnitPrices where
+    build (ExecutionUnitPrices perStep perMem) =
+        build $ show perStep <> " per step, " <> show perMem <> " per memory unit"
+
+instance ToJSON ExecutionUnitPrices where
+    toJSON ExecutionUnitPrices{priceExecutionSteps, priceExecutionMemory} =
+        object [ "step_price"  .= toRationalJSON priceExecutionSteps
+               , "memory_unit_price" .= toRationalJSON priceExecutionMemory
+               ]
+     where
+         toRationalJSON :: Rational -> Value
+         toRationalJSON r = case fromRationalRepetendLimited 20 r of
+             Right (s, Nothing) -> toJSON s
+             _                  -> toJSON r
```

Comment:
Hm, are we sure this dual representation is a good idea?

As a user looking at the value, having a simple scientific representation does seem nice. 

As an application integrating with the wallet API you always need to care about the rational case anyway, so the dual representations makes the JSON parsing more difficult.

But perhaps this aids downstream applications to deal with rationals correctly, if there is no easy `fromRationalRepetendLimited` in their language? E.g. when Daedalus wants to display this to the user.

Reply:
so from protocol parameters we got
```
executionUnitPrices = Just (ExecutionUnitPrices {priceExecutionSteps = 1 % 1, priceExecutionMemory = 1 % 1 )
```
and I differentiate between two cases when we have remainder = 0 and otherwise when ratio is evaluated. And we have
1. when rational is integer then number
2. when rational is not integer then denominator/nominator

Examples:
```
0.2 -> 1/5 -> denominator:5, nominator:1
2 -> 2
```

And I would slightly;-) argue it is acceptable solution. But I am open to other things. Not very strong opinion (as always from me when it comes to user facing info)

Reply:
May I just plainly ask though: what was your motivation for
```
0.2 -> 1/5 -> denominator:5, nominator:1
2 -> 2
```
rather than
```
0.2 -> 1/5 -> denominator:5, nominator:1
2 -> denominator:1, nominator:2
```
?

Reply:
just thought when we have integer it is redundant to have denominator. That was my thinking...also I see node team uses the same approach for presenting rational so maybe it is good to consistent with them (with balance tx endpoint we accept start to accept some cardano-api's types in our api as a move for better interoperability)